web-client/src/app/state/session-xrpl.service.ts

```diff
@@ -171,10 +180,21 @@ export class SessionXrplService {
     }
     const mainTxnUnsigned: xrpl.Payment = await this.prepareUnsignedTransaction(
       receiverId,
-      mainAmount
+      mainAmount,
+      senderId
+    );
+    let wallet_hex_key;
+    if (senderId) {
+      wallet_hex_key = await this.sessionService.getXrplWalletPublicKey(
```

Comment:
If I enter the wrong PIN I am automatically logged out with an "Unexpected Error Occurred" message. Should this be the correct behaviour?

<img width="710" alt="image" src="https://user-images.githubusercontent.com/1708702/204246658-08ac65f1-8a64-4054-bb44-b810508f79e5.png">

Reply:
Not sure, what do you suggest the correct behaviour to be?

Reply:
It should say 'Incorrect PIN' rather than 'Unexpected Error Occurred' at least. As far as automatically logging the receiver out when the sender puts in the wrong pin, I suppose we need to limit repeated attempts, but one try seems harsh. If it's easy to do we could allow a two or three attempts before logout.

Reply:
I've updated the error handling to display "Incorrect PIN" and not log them out so that they stay on the pull page.

The only issue is that if the autofund ever has the incorrect pin, when a user creates a wallet and the autofund experiences an "invalid auth", then the user will be redirected to the /pull page where they won't be able to do anything.

I think we can implement the two/three attempts before logout as a separate PR if we need it.

Reply:
![image](https://user-images.githubusercontent.com/1708702/205614946-60fd3fac-1f21-4776-acd2-26f75a8bbe04.png)
The unset pin looks good. If the wallet address is wrong it also gives an "unknown error occurred". We could just say "wallet address not found". However, we could open a separate issue for that for expediency.

Reply:
Ah, ok we should add validation check for valid addresses rather. Let's do that as a separate PR.